docs/commcare-cloud/deploy-branches.md

```diff
@@ -0,0 +1,100 @@
+# Deploying specific branches
+
+In some cases it may be desirable to have full control over what code is deployed. This can
+be accomplished by creating a `deploy_branches.yml` configuration file for you environment.
+This file defined exactly which branches will get deployed.
+
+The format of the file is as follows:
+
+```yaml
+trunk: master
+name: autostaging
+branches:
+  - feature1
+  - feature2
+  - forkowner:feature3 # branch from fork of repository
+submodules:
+  submodules/module1:
+    branches:
+      - feature1
+      - forkowner:feature2 # branch from fork of repository
+  submodules/module2:
+    trunk: develop
+    branches:
+      - feature2
+```
+When not specified, a submodule's trunk and name inherit from the parent.
+
+Multiple repositories can also be specified:
+
+```yaml
+   dimagi/commcare-hq:  # this repo is required
```

Comment:
This will still only operate on the main HQ repo, right?  Like, prospective users will need write access to that for this to work?

Reply:
there's no check to see that it's actually the HQ repo so I guess it could be any repo - but that wouldn't really make much sense since deploy is tied to the HQ repo.

Yes, you would need write access to push the branches.